Documentation/architecture.md

```diff
@@ -0,0 +1,70 @@
+# Rocket architecture
+
+## Overview
+
+Rocket consists only of a command-line tool, `rkt`, and does not have a daemon. All state is communicated via the filesystem. Facilities like file-locking are used to ensure co-operation and mutual exclusion between concurrent invocations of the `rkt` command.
+
+## Stages
+
+Execution with Rocket is divided into several distinct stages.
+
+### Stage 0
+
+The first stage is the actual `rkt` binary itself. When running a container, this binary is responsible for performing a number of initial preparatory tasks:
+- Fetching the specified ACIs, including the stage 1 ACI of --stage1-image if specified.
+- Generating a Container UUID
+- Generating a Container Runtime Manifest
+- Creating a filesystem for the container
+- Setting up stage 1 and stage 2 directories in the filesystem
+- Unpacking the stage 1 ACI into the container filesystem
+- Unpacking the ACIs and copying each app into the stage2 directories
+
+Given a run command such as:
+
+```
+[~/rocket-v0.2.0]$ sudo ./rkt run \
+	sha512-8a30f14877cd8065939e3912542a17d1a5fd9b4c \
+	sha512-abcd29837d89389s9d0898ds908ds890df890908
+```
+
+a container manifest compliant with the ACE spec will be generated, and the filesystem created by stage0 should be:
+
+TODO(jonboulle): layout needs to be updated to reflect ACIs-as-stage-1
+
+```
+/container
+/stage1
```

Comment:
Each directory shold end with a `/` to more clearly show it is a directory and not a file:

`/stage1/`